### PR TITLE
Single page app

### PR DESCRIPTION
**Pull request by will-moore**
Add a single page app, to allow the centre panel to be used for filtering search results.
Top link in webclient "Parade" opens parade app in new window.
FullText search for images allows the centre panel filtering etc to work with search results (showing that we can use centre panel without it needing jstree etc).
Very bare-bones currently, but working:

screen shot 2018-02-27 at 15 00 33

### CODE REVIEW

File: omero_parade/templates/omero_parade/index.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    </head>
+
+    <body>
+        <div id="full_page_app"></div>
+    </body>
+
+    <script>
+        omero_parade.full_page_app("full_page_app");
+    </script>
+</html>
```
Comment (jburel) [inline omero_parade/templates/omero_parade/index.html]: add extra line

File: src/js/filter/FilterContainer.js [hunk 2]
```diff
@@ -13,8 +13,14 @@ export default React.createClass({
     componentDidMount: function() {
         // list available filters (TODO: only for current data? e.g. plate)
         let url = window.PARADE_FILTERS_URL;
-        if (this.props.plateId) url += '?plate=' + this.props.plateId;
-        if (this.props.datasetId) url += '?dataset=' + this.props.plateId;
+        if (this.props.plateId) {
+            url += '?plate=' + this.props.plateId;
+        }
+        else if (this.props.datasetId) {
+            url += '?dataset=' + this.props.plateId;
```
Comment (jburel) [inline src/js/filter/FilterContainer.js]: datasetId

File: src/js/filter/ParadeFilter.js [hunk 3]
```diff
@@ -19,6 +19,17 @@ export default React.createClass({
...
         if (this.props.plateId) url += '?plate=' + this.props.plateId;
         if (this.props.fieldId !== undefined) url += '&field=' + this.props.fieldId;
+        if (this.props.plateId) {
+            url += '?plate=' + this.props.plateId;
+            if (this.props.fieldId !== undefined) {
+                url += '&field=' + this.props.fieldId;
+            }
+        }
+        else if (this.props.datasetId) {
+            url += '?dataset=' + this.props.plateId;
```
Comment (jburel) [inline src/js/filter/ParadeFilter.js]: datasetId

File: src/js/search/SearchApp.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    render: function() {
+        return (
+            <div className="columnContainer">
+                <div className="leftPanel">
+                    <SearchForm 
+                        setSearchResults={this.setSearchResults} />
+                </div>
+                <div className="paradeCentrePanel">
+                    <FilterHub
+                        // datasetId={this.props.parentNode.data.obj.id}
```
Comment (jburel) [inline src/js/search/SearchApp.js]: remove

**Comment by jburel** [comment]
Maybe include commits from https://github.com/will-moore/omero-parade/pull/8 in this PR to avoid conflicts

File: src/js/filter/FilterContainer.js (hunk 2 above)
Comment (will-moore) [reply src/js/filter/FilterContainer.js]: Why datasetId? We don't tend to use *Id in query strings elsewhere.

Comment (will-moore) [reply src/js/filter/FilterContainer.js]: Ooops - sorry. I see it.

**Pull request merged**